hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroHFileReader.java

```diff
@@ -333,10 +329,13 @@ public IndexedRecord next() {
 
   private static Option<IndexedRecord> fetchRecordByKeyInternal(HFileScanner scanner, String key, Schema writerSchema, Schema readerSchema) throws IOException {
     KeyValue kv = new KeyValue(key.getBytes(), null, null, null);
-    if (scanner.seekTo(kv) != 0) {
+    int returnVal = scanner.reseekTo(kv);
+    if (returnVal != 0) {
+      // key is not found.
```

Comment:
Just replace `seekTo` with `reseekTo` should be fine?

Reply:
yes, will fix it.